src/eos-dropbox-app

```diff
@@ -235,6 +243,25 @@ class DropboxLauncher():
             logging.info("Not the main launcher instance. Exiting")
             self._quit()
 
+    # This method is required for compatibility with xdg-desktop-portal < 0.6, for
+    # installations that might have updated the DropBox app before the base OS.
+    def _fallbackToOpenURIMethod(self, path):
+        handle = self._proxy.OpenURI('(ssa{sv})',
+                                     '',                          # Parent window ID
+                                     "file://{}".format(path),    # URI
+                                     None) # Options
+
+        # OpenURI methods returns a handle to a 'request object', which stays
+        # alive for the duration of the user interaction related to the method
+        # call, so we connect to its Response signal to know when it's all over.
+        self._bus.signal_subscribe('org.freedesktop.portal.Desktop',
```

Comment:
This code is duplicated in the two functions – not ideal.

Reply:
Not at all, sorry about that :)